lib/connection.js

```diff
@@ -112,15 +113,26 @@ class Connection extends EventEmitter {
   }
 
   // Send a handshake packet over the connection
-  //   appName - name of an application to connect to
+  //   app - string or object, application to connect to as 'name' or
+  //         'name@version' or { name, version }
   //   login - user name (optional)
   //   password - user password (optional)
   //   callback - callback function to invoke after the handshake is completed
   //
-  handshake(appName, login, password, callback) {
+  handshake(app, login, password, callback) {
+    let target;
+    if (typeof app === 'string') {
+      const [name, version] = app.split('@');
+      if (semver.valid(version)) target = [name, version];
+      else target = name;
+    } else if (!app.version) {
+      target = app.name;
+    } else {
+      target = [app.name, app.version];
```

Comment:
@aqrln, as far as I understand, first option you provided isn't possible, because it is always an array, check out `_createPacket` function. Sometimes version can just be omitted in the array, meaning that latest version should be used, and packet will look like this:
`{ handshake: [0, 'app'] }`.